common/src/main/java/com/wynntils/screens/partymanagement/PartyManagementScreen.java

```diff
@@ -0,0 +1,426 @@
+
+            suggestedPlayersWidgets.add(new SuggestionPlayer(
+                    this.width / 2 + 204, this.height / 2 - 125 + i * 20 - 10, 200, 20, playerName));
+        }
+    }
+
+    @SubscribeEvent
+    public void onFriendJoin(FriendConnectionEvent.Join e) {
+        reloadSuggestedPlayersWidgets();
+    }
+
+    @SubscribeEvent
+    public void onFriendLeave(FriendConnectionEvent.Leave e) {
+        // Delay the reload to allow the scoreboard to update
+        Managers.TickScheduler.scheduleLater(this::reloadSuggestedPlayersWidgets, 5);
+    }
+
+    /**
+     * Reloads the list of party members widgets
+     * <p>
+     * This should be called when the party list is updated or when the refresh button is pressed
+     */
+    private void reloadMembersWidgets() {
+        partyMembersWidgets.clear();
+        List<String> partyMembers = new ArrayList<>(Models.Party.getPartyMembers());
+        for (int i = 0; i < partyMembers.size(); i++) {
+            String playerName = partyMembers.get(i);
+            if (playerName == null) continue;
+
+            partyMembersWidgets.add(new PartyPlayer(
+                    this.width / 2 - X_START + 4,
+                    this.height / 2 - 125 + i * 20 - 10,
+                    200,
+                    20,
+                    playerName,
+                    offlineMembers.contains(playerName)));
+        }
+    }
+
+    @SubscribeEvent
+    public void onPartyUpdate(RelationsUpdateEvent.PartyList e) {
+        reloadMembersWidgets();
+        reloadSuggestedPlayersWidgets(); // Reload because we don't want to suggest party members
+    }
+
+    private void inviteFromField() {
+        // Remove all except commas, semicolons, whitespaces, and characters possible in name
+        String fieldText =
+                INVITE_REPLACER.matcher(inviteInput.getTextBoxInput()).replaceAll("");
+        fieldText = COMMA_REPLACER.matcher(fieldText).replaceAll(","); // semicolons and spaces to comma
+        if (fieldText.isBlank()) return;
+
+        if (!Models.Party.isPartying()) {
+            createParty();
+        }
+
+        Set<String> toInvite = new HashSet<>(List.of(fieldText.split(",")));
+        toInvite.removeAll(Models.Party.getPartyMembers());
+        toInvite.forEach(playerName -> McUtils.sendCommand("party invite " + playerName));
+
+        inviteInput.setTextBoxInput("");
+    }
+
+    private void refreshParty() {
+        Models.Party.requestPartyListUpdate();
+        reloadSuggestedPlayersWidgets();
+        reloadMembersWidgets();
+        refreshOffline();
+    }
+
+    private void kickOffline() {
+        refreshParty();
+        offlineMembers.forEach(playerName -> McUtils.sendCommand("party kick " + playerName));
+    }
+
+    private void refreshOffline() {
+        offlineMembers.clear();
+        offlineMembers.addAll(Models.Party.getPartyMembers());
+        offlineMembers.removeAll(McUtils.mc().level.getScoreboard().getTeamNames());
```

Comment:
This line needs a bit of explaining.

Reply:
All the members in the party are added to offlineMembers, then the ones that have a scoreboard entry are removed. Scoreboard entry only exists for those that are online in the server.

I will also add this as a comment.